gcloud-java-core/src/main/java/com/google/gcloud/IamPolicy.java

```diff
@@ -0,0 +1,440 @@
+     *     legacy role is "viewer". An example of a valid service-specific role is
+     *     "pubsub.publisher".
+     * @param members list of identities associated with the role.
+     */
+    public static Acl of(String role, List<Identity> members) {
+      return new Acl(new Builder(role).identities(members));
+    }
+
+    /**
+     * Returns an ACL for the role type and identities provided.
+     *
+     * @param role string representing the role, without the "roles/" prefix.  An example of a valid
+     *     legacy role is "viewer". An example of a valid service-specific role is
+     *     "pubsub.publisher".
+     * @param first identity associated with the role.
+     * @param others any other identities associated with the role.
+     */
+    public static Acl of(String role, Identity first, Identity... others) {
+      return new Acl(new Builder(role).addIdentity(first, others));
+    }
+
+    public Builder toBuilder() {
+      return new Builder(role).identities(identities);
+    }
+
+    @Override
+    public int hashCode() {
+      return Objects.hash(identities, role);
+    }
+
+    @Override
+    public boolean equals(Object obj) {
+      if (!(obj instanceof Acl)) {
+        return false;
+      }
+      Acl other = (Acl) obj;
+      return Objects.equals(identities, other.identities()) && Objects.equals(role, other.role());
+    }
+  }
+
+  /**
+   * Builder for an IAM Policy.
+   */
+  public static class Builder {
+
+    private final List<Acl> acls = new LinkedList<>();
+    private String etag;
+    private int version;
+
+    /**
+     * Replaces the builder's list of ACLs with the given list of ACLs.
+     */
+    public Builder acls(List<Acl> acls) {
+      this.acls.clear();
+      this.acls.addAll(acls);
+      return this;
+    }
+
+    /**
+     * Adds one or more ACLs to the policy.
+     */
+    public Builder addAcl(Acl first, Acl... others) {
+      acls.add(first);
+      acls.addAll(Arrays.asList(others));
+      return this;
+    }
+
+    /**
+     * Removes the specified ACL.
+     */
+    public Builder removeAcl(Acl acl) {
+      acls.remove(acl);
+      return this;
+    }
+
+    /**
+     * Sets the policy's etag.
+     *
+     * <p>Etags are used for optimistic concurrency control as a way to help prevent simultaneous
+     * updates of a policy from overwriting each other. It is strongly suggested that systems make
+     * use of the etag in the read-modify-write cycle to perform policy updates in order to avoid
+     * race conditions. An etag is returned in the response to getIamPolicy, and systems are
+     * expected to put that etag in the request to setIamPolicy to ensure that their change will be
+     * applied to the same version of the policy.  If no etag is provided in the call to
+     * setIamPolicy, then the existing policy is overwritten blindly.
+     */
+    public Builder etag(String etag) {
```

Comment:
Actually that would be an issue when this is in a shared/common place.

One option, just thinking out loud, is to make this class abstract and the builder protected.
That mean each service that supports IAM will provide its own subclass (with its own to/from pb)
and can even using generics for the role type (so each service can have its own roles enum if it has such). In this case etag can be protected.